CONTRIBUTING.md

```diff
@@ -9,6 +9,39 @@ Druxt is an open source project, built, supported and maintained by the communit
 
 * * *
 
+## Development Environment Setup
```

Comment:
@captaindav I moved this section up to the top, as I believe it deserves priority for a Contributing guide, and I also updated the text.

I removed the steps regarding feature branching and such, as that is covered under the Pull requests section.